### PR TITLE
Add I/O operation speed decorator

### PR DESCRIPTION
**Pull request by xakep666**
Unit variants: KiB, kB, no unit
Example gifs:
mpb-multi
mpb-single

### CODE REVIEW

**Comment by vbauerster** [comment]
[Here](https://github.com/vbauerster/getparty/blob/master/gp/decorator.go#L41) I'm using speed decorator with simple speed calculation algorithm, it's not part of mpb though. What is a benefit for having speed decorator with complex calculation, like the one you propose?

**Comment by xakep666** [comment]
I wanted to show instantaneous speed but you method is better because it`s simple and not requires filtering. I`ll re-push comits with it.

File: decor/decorators.go [hunk 1]
```diff
@@ -266,3 +266,84 @@ func CalcPercentage(total, current, width int64) (perc int64) {
...
+//	`unitFormat` printf compatible verb for value, like "%f" or "%d"
+//
+//	`width` width reservation to apply, ignored if `DwidthSync` bit is set
+//
+//	`conf` bit set config, [DidentRight|DwidthSync|DextraSpace]
+//
+// unitFormat example:
+//
+//	"%.1f" = "1.0" or "% .1f" = "1.0"
+func SpeedNoUnit(unitFormat string, minWidth, conf int) DecoratorFunc {
```
Comment (vbauerster) [inline decor/decorators.go]: Can you please rename `minWidth` to `width`, so godoc comment matches actual param name.

File: decor/decorators.go [hunk 2]
```diff
@@ -266,3 +266,84 @@ func CalcPercentage(total, current, width int64) (perc int64) {
...
+//	`unitFormat` printf compatible verb for value, like "%f" or "%d"
+//
+//	`width` width reservation to apply, ignored if `DwidthSync` bit is set
+//
+//	`conf` bit set config, [DidentRight|DwidthSync|DextraSpace]
+//
+// unitFormat example:
+//
+//	"%.1f" = "1.0MB/s" or "% .1f" = "1.0 MB/s"
+func SpeedKibiByte(unitFormat string, minWidth, conf int) DecoratorFunc {
```
Comment (vbauerster) [inline decor/decorators.go]: Can you please rename `minWidth` to `width`, so godoc comment matches actual param name.

File: decor/decorators.go [hunk 3]
```diff
@@ -266,3 +266,84 @@ func CalcPercentage(total, current, width int64) (perc int64) {
...
+//	`unitFormat` printf compatible verb for value, like "%f" or "%d"
+//
+//	`width` width reservation to apply, ignored if `DwidthSync` bit is set
+//
+//	`conf` bit set config, [DidentRight|DwidthSync|DextraSpace]
+//
+// unitFormat example:
+//
+//	"%.1f" = "1.0MB/s" or "% .1f" = "1.0 MB/s"
+func SpeedKiloByte(unitFormat string, minWidth, conf int) DecoratorFunc {
```
Comment (vbauerster) [inline decor/decorators.go]: Can you please rename `minWidth` to `width`, so godoc comment matches actual param name.

File: decor/decorators.go [hunk 4]
```diff
@@ -266,3 +266,84 @@ func CalcPercentage(total, current, width int64) (perc int64) {
...
+//
+//	`unitFormat` printf compatible verb for value, like "%f" or "%d"
+//
+//	`width` width reservation to apply, ignored if `DwidthSync` bit is set
+//
+//	`conf` bit set config, [DidentRight|DwidthSync|DextraSpace]
+//
+// unitFormat example:
+//
+//	"%.1f" = "1.0MB/s" or "% .1f" = "1.0 MB/s"
```
Comment (vbauerster) [inline decor/decorators.go]: `"%.1f" = "1.0MiB/s" or "% .1f" = "1.0 MiB/s"` would be a more relevant.

**Comment by xakep666** [comment]
@vbauerster made requested changes

**Comment by vbauerster** [comment]
Благодарю!

**Pull request merged**